Copyright (c) 1997-2014 University of Cambridge
Copyright (c) 2007-2012, Google Inc.
Copyright (c) 2009-2014 Zoltan Herczeg

Redistribution and use in source and binary forms, with or without
modification, are permitted provided that the following conditions are met:

    * Redistributions of source code must retain the above copyright notice,
      this list of conditions and the following disclaimer.

    * Redistributions in binary form must reproduce the above copyright
      notice, this list of conditions and the following disclaimer in the
      documentation and/or other materials provided with the distribution.

    * Neither the name of the University of Cambridge nor the name of Google
      Inc. nor the names of their contributors may be used to endorse or
      promote products derived from this software without specific prior
      written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
POSSIBILITY OF SUCH DAMAGE.

Date: Apr 21 10:49:44 2017

From: aa010cd852e536884909037b484a3d5aabebb1cf May 26 11:36:26 2017

Subject: [PATCH] RDK-17468, RDKB-8940: Open Source Updates

	Upgrade libpcre from 8.34 to 8.39 as PREFERRED_VERSION
        to address known high risk security vulnerabilities.

	This patch address a namespace collision with libc.

	Although there is no "#include <regex.h>" in the source file, at
	runtime, it's unintentionally linked to the libc version, the regcomp of
	libc is called instead the pcre one using pcre's data structure...
	that looks like a disaster.

	Can patch is from Debian (and Ubuntu 11.04alpha has it also).

Source: OPENEMBEDDED

Upstream-Status: Submitted [ http://cgit.openembedded.org/openembedded-core/commit/meta/recipes-support/libpcre/libpcre/fix-pcre-name-collision.patch?h=pyro&id=32b2e20e7f6484830bac0510414b1950abbc1a96 ]

Signed-off-by: Qing He <qing.he@intel.com>
Signed-off-by: Saul Wold <sgw@linux.intel.com>
---


--- a/pcreposix.h	2010-05-17 00:17:23.000000000 +0800
+++ b/pcreposix.h	2009-01-15 04:32:17.000000000 +0800
@@ -133,14 +130,19 @@
 
 /* The functions */
 
-PCREPOSIX_EXP_DECL int regcomp(regex_t *, const char *, int);
-PCREPOSIX_EXP_DECL int regexec(const regex_t *, const char *, size_t,
+PCREPOSIX_EXP_DECL int pcreposix_regcomp(regex_t *, const char *, int);
+PCREPOSIX_EXP_DECL int pcreposix_regexec(const regex_t *, const char *, size_t,
                      regmatch_t *, int);
-PCREPOSIX_EXP_DECL size_t regerror(int, const regex_t *, char *, size_t);
-PCREPOSIX_EXP_DECL void regfree(regex_t *);
+PCREPOSIX_EXP_DECL size_t pcreposix_regerror(int, const regex_t *, char *, size_t);
+PCREPOSIX_EXP_DECL void pcreposix_regfree(regex_t *);
 
 #ifdef __cplusplus
 }   /* extern "C" */
 #endif
 
+#define regcomp pcreposix_regcomp
+#define regexec pcreposix_regexec
+#define regerror pcreposix_regerror
+#define regfree pcreposix_regfree
+
 #endif /* End of pcreposix.h */
